### ISSUE TITLE
Unable to write to Portal.

### ISSUE THREAD

**Issue by cuthred** [silicontrip/SkyReader#8]
Trying to use this on Linux and get the following error (portal from Spyro Adventures, 1430:0150)

"Unable to write to Portal."

Permissions on hidraw seem to be ok:
crw-rw-rw- 1 root root 249, 4 Sep 28 09:45 /dev/hidraw4

This happens in RestartPortal, as hid_write returns -1. libexplain gives the following output for write syscall:

write(fildes = 3 "/dev/hidraw4", data = 0x7FFFB369C890, data_size = 33) failed, Protocol error (71, EPROTO)

Is there more documentation on the portal protocols?

**Comment by silicontrip** [comment]
unfortunately I don't know much about the portal protocol.  I wrote a new UI and usb integration for an existing dos program. 
I also focussed mostly on an OSX port, so I'm not too familiar how usb devices are handled in the linux kernel.  Other people have made ports for linux.  You may have more luck with those.

That error, however, looks like it's being thrown at the usb or hid layer, not from the portal. The portal will normally respond with errors inbound.  A write would not fail.
This error looks like it was generated by the HID library.  Perhaps tracing the HID messages might help.

**Comment by cuthred** [comment]
yes, the error comes from the hid device driver which calls the syscall write to send data to the portal. The linux port from reedstrm seems to be dead.
